test/TelfordDestination.spec.js

```diff
@@ -0,0 +1,86 @@
+// const { EtherscanProvider } = require("@ethersproject/providers");
+const { use, expect } = require("chai");
+const { ethers, waffle } = require("hardhat");
+// const { isCallTrace } = require("hardhat/internal/hardhat-network/stack-traces/message-trace");
+
+describe("TelfordDestination", async () => {
+    let telfordDestination;
+    let owner, bonder, user, randomEOA;
+
+    let bridgeRequestId = 1;
+    let depositAmount = ethers.utils.parseEther("1");
+    
+    beforeEach("deploy contract", async () => {
+        // get accounts using signers
+        [owner, bonder, user, randomEOA] = await ethers.getSigners();
+
+        // get contract using ContractFactory
+        const TelfordDestination = await ethers.getContractFactory("TelfordDestination");
+
+        // deployed instance
+        telfordDestination = await TelfordDestination.deploy(
+            bonder.address,
+            "0x4200000000000000000000000000000000000007",
+            "0xb5cf88Df79CfdFe52572C5F015017fe486979b61"
+        );
+
+        await telfordDestination.deployed();
+    });
+
+    describe("depositAndDistribute", () => {
+
+        it("should revert when not called by registered bonder", async () => {
+            const non_bonder_deposit_tx = telfordDestination.depositAndDistribute(user.address, bridgeRequestId, {value: depositAmount });
+
+            await expect(non_bonder_deposit_tx)
+                .to.be.revertedWith("You must be a bonder to initiate withdrawAndDistribute!");
+        });
+
+         it("should successfully deposit depositAmount from bonder to user", async () => {
+            const contractAddress = telfordDestination.address;
+            
+            // pre-tx balances
+            const bonderPrevBalance = await bonder.getBalance();
+            const contractPrevBalance = await ethers.provider.getBalance(contractAddress);;
+            const userPrevBalance = await bonder.getBalance();
```

Comment:
I just noticed, you're setting the bonder balance here, not the user balance

Reply:
Ah yeah, copy paste error, will fix.